server/jdkhttp-server/src/main/scala/sttp/tapir/server/jdkhttp/internal/KMPMatcher.scala

```diff
@@ -0,0 +1,49 @@
+package sttp.tapir.server.jdkhttp.internal
+
+import scala.collection.mutable
+
+class KMPMatcher(delimiter: Array[Byte]) {
```

Comment:
one final comment here, sorry :) maybe it would be worth adding some unit tests for `KMPMatcher` to check for some corner cases and "happy paths"? I know there's an additional multipart test, but there might be some data combinations which aren't covered. And I'm not that familiar with the algorithm to see that it's correct ;)

Reply:
Don't be sorry to have requirements that makes the code better :smile: 
I'll add some tests!